app/src/main/java/org/simple/clinic/editpatient/deletepatient/DeletePatientScreen.kt

```diff
@@ -0,0 +1,124 @@
+import io.reactivex.subjects.PublishSubject
+import kotlinx.android.synthetic.main.screen_delete_patient.view.*
+import org.simple.clinic.R
+import org.simple.clinic.di.injector
+import org.simple.clinic.home.HomeScreenKey
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.patient.DeletedReason
+import org.simple.clinic.router.screen.RouterDirection
+import org.simple.clinic.router.screen.ScreenRouter
+import org.simple.clinic.util.unsafeLazy
+import org.simple.clinic.widgets.DividerItemDecorator
+import org.simple.clinic.widgets.dp
+import javax.inject.Inject
+
+class DeletePatientScreen(context: Context, attrs: AttributeSet) : ConstraintLayout(context, attrs), UiActions, DeletePatientUi {
+
+  @Inject
+  lateinit var screenRouter: ScreenRouter
+
+  @Inject
+  lateinit var effectHandlerFactory: DeletePatientEffectHandler.Factory
+
+  private val viewRenderer = DeletePatientViewRenderer(this)
+
+  private val screenKey by unsafeLazy {
+    screenRouter.key<DeletePatientScreenKey>(this)
+  }
+
+  private val events = PublishSubject.create<DeletePatientEvent>()
+
+  private val delegate by unsafeLazy {
+    MobiusDelegate.forView(
+        events = events,
+        defaultModel = DeletePatientModel.default(screenKey.patientUuid),
+        init = DeletePatientInit(),
+        update = DeletePatientUpdate(),
+        effectHandler = effectHandlerFactory.create(this).build(),
+        modelUpdateListener = viewRenderer::render
+    )
+  }
+
+  @SuppressLint("CheckResult")
+  override fun onFinishInflate() {
+    super.onFinishInflate()
+    if (isInEditMode) return
+
+    context.injector<DeletePatientScreenInjector>().inject(this)
+
+    toolbar.setNavigationOnClickListener { screenRouter.pop() }
+    with(deleteReasonsRecyclerView) {
+      addItemDecoration(DividerItemDecorator(context, marginStart = 56.dp, marginEnd = 16.dp))
+    }
+  }
+
+  override fun onAttachedToWindow() {
+    super.onAttachedToWindow()
+    delegate.start()
+  }
+
+  override fun onDetachedFromWindow() {
+    delegate.stop()
+    super.onDetachedFromWindow()
+  }
+
+  override fun onSaveInstanceState(): Parcelable? {
+    return delegate.onSaveInstanceState(super.onSaveInstanceState())
+  }
+
+  override fun onRestoreInstanceState(state: Parcelable?) {
+    val viewState = delegate.onRestoreInstanceState(state)
+    super.onRestoreInstanceState(viewState)
+  }
+
+  override fun showDeleteReasons(patientDeleteReasons: List<PatientDeleteReason>, selectedReason: PatientDeleteReason?) {
+  }
+
+  override fun showConfirmDeleteDialog(patientName: String, deletedReason: DeletedReason) {
+    val title = when (deletedReason) {
+      DeletedReason.Duplicate -> context.getString(R.string.deletereason_duplicate_patient_confirm_title)
+      DeletedReason.AccidentalRegistration -> context.getString(R.string.deletereason_accidental_registration_confirm_title)
+      is DeletedReason.Unknown -> deletedReason.actualValue
```

Comment:
Since we will never show the deleted reason in the UI for an already deleted patient, we might as well throw an `IllegalArgumentException` here instead of showing the non-human readable value. What do you think?

Reply:
We are throwing it when we read the message anyway.

Reply:
Pragati mentioned about having a common title and message for the confirmation dialog. I will confirm that and make that change.